### PR TITLE
The architecture of host pc must be same architecture with target.

### PR DESCRIPTION
**Pull request by leemgs**
Let's make exception handling for different architecture between host pc and target board.

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

### CODE REVIEW

**Comment by leemgs** [comment]
\cc @jkotas , @benpye 

Here is  command to evaluate for Linux/ARM. This patch is to resolve the below  issue. 

```
geunsik@u1404:/work/dotnet/arm-32/coreclr$ time ./build.sh arm debug clean verbose
          . . . OMISSION . . . . 
Generating native image for mscorlib.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Could not load file or assembly '/work/dotnet/arm-32/coreclr/bin/Product/Linux.arm.Debug/mscorlib.dll' or one of its dependencies. An attempt was made to load a program with an incorrect format.
 (Exception from HRESULT: 0x8007000B (COR_E_BADIMAGEFORMAT))
Error: compilation failed for "/work/dotnet/arm-32/coreclr/bin/Product/Linux.arm.Debug/mscorlib.dll" (0x8007000b)
Failed to generate native image for mscorlib.

real	12m59.018s
user	40m44.844s
```
sys	3m23.816s
geunsik@u1404:/work/dotnet/arm-32/coreclr$

**Comment by jkotas** [comment]
All Unix builds are failing with:

```
System.IO.FileNotFoundException: NuGetResources.PackageAuthoring_FileNotFound /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/bin/Product/Linux.x64.Checked/mscorlib.ni.dll
```

Why is the crosscompile condition for this?

**Comment by leemgs** [comment]
> All Unix builds are failing with:
> System.IO.FileNotFoundException: NuGetResources.PackageAuthoring_FileNotFound 
> /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/bin/Product/Linux.x64.Checked/mscorlib.ni.dll

Thanks your fast feedback. This error is normal. I will resend PR after fixing this issue. 



> Why is the $__CrossBuild condition not sufficient?

`if [ $__CrossBuild != 1 ]; then statement `is to skip the generation step to make `mscorlib.ni.dll` from `mscorlib.dll.`
As a next step, This patch is to handle` "time ./build.sh arm debug clean verbose"` command in case that  generate only mscorlib.dll for Linux/ARM  as well as the existing case of `if [ $__CrossBuild != 1 ]`
In this case, we need to add additional if statement because the architecture of host pc must be same architecture with target.

**Comment by leemgs** [comment]
\cc @jkotas , @benpye ,  @myungjoo 

We have submitted  updated PR v2  with "squash" command for a single commit after fixing issue of the existing PR.

File: build.sh [hunk 1]
```diff
@@ -251,17 +263,27 @@ build_mscorlib()
...
-           echo "Generating native image for mscorlib."
-           $__BinDir/crossgen $__BinDir/mscorlib.dll
-           if [ $? -ne 0 ]; then
-               echo "Failed to generate native image for mscorlib."
-               exit 1
-           fi
+       # The architecture of host pc must be same architecture with target.
+       if [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x64" ) ]]; then
+           build_mscorlib_ni
+       elif [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x86" ) ]]; then
```
Comment (benpye) [inline build.sh]: Does this and the arm64 -> arm build actually work?

File: build.sh [hunk 2]
```diff
@@ -251,17 +263,27 @@ build_mscorlib()
...
     if [ $__CrossBuild != 1 ]; then
-       if [ $__SkipCoreCLR == 0 -a -e $__BinDir/crossgen ]; then
-           echo "Generating native image for mscorlib."
-           $__BinDir/crossgen $__BinDir/mscorlib.dll
-           if [ $? -ne 0 ]; then
-               echo "Failed to generate native image for mscorlib."
-               exit 1
-           fi
+       # The architecture of host pc must be same architecture with target.
+       if [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x64" ) ]]; then
```
Comment (jkotas) [inline build.sh]: Can this be simplified to `if [[ $__HostArch" == "$__BuildArch" ]]`?

File: build.sh (hunk 1 above)
Comment (leemgs) [inline build.sh]: Yes. We sometimes have built arm binary on arm64-based host for embedded devices.

File: build.sh (hunk 2 above)
Comment (leemgs) [inline build.sh]: We have made a simple condition at first as you described. We have updated the simple condition to handle both 32bit and 64bit architecture.

Comment (jkotas) [inline build.sh]: I would be simpler and easier to follow if there is the simple exact match first, and then the additional mismatched combinations you would like to allow after it. Something like:

```
if [[ $__HostArch" == "$__BuildArch" ]];
    build_mscorlib_ni;
elif [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x86" ) ]]; 
    build_mscorlib_ni;
elif [[ ( "$__HostArch" == "arm64" ) && ( "$__BuildArch" == "arm" ) ]]; 
    build_mscorlib_ni;
```

**Comment by leemgs** [comment]
@jkotas , We have applied your comment. Thanks.

**Comment by jkotas** [comment]
LGTM

**Pull request merged**